src/dune_manager/dune_manager.ml

```diff
@@ -424,10 +426,33 @@ module Client = struct
   type t =
     { socket : out_channel
     ; fd : Unix.file_descr
+    ; input : char Stream.t
     ; memory : Dune_memory.Memory.t
+    ; thread : Thread.t
     }
 
-  let make () =
+  type command = Dedup of (Path.Build.t * Path.t)
+
+  let pp_command fmt = function
```

Comment:
We usually write `to_dyn : t -> Dyn.t` functions. Then, when we need to pretty print something we use `Dyn.pp`.